lib/ts/RbTree.cc

```diff
@@ -272,7 +272,7 @@ namespace detail
 
       while (parent) { // @a n is not the root
         // If the current node is RED, we can just recolor and be done
-        if (n == RED) {
+        if (n && n == RED) {
```

Comment:
What's happening here? `n` should be a pointer but this line looks like comparing a pointer and an enum value. Shouldn't it be `n->_color == RED` ?

Reply:
There is an overload where if `n` is `nullptr` it compares equal to `BLACK`.